react/Button/test/button.spec.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+/* eslint-env jest */
+
+import React from 'react'
+import { configure, shallow } from 'enzyme'
+import Adapter from 'enzyme-adapter-react-15'
+
+import { Button } from '../'
+import Icon from '../../Icon'
+
+configure({ adapter: new Adapter() })
+
+describe('Button', () => {
+  ;['', 'subtle'].forEach(variant => {
+    describe(`${Button.name} component${variant &&
+      ' (' + variant + ' variant)'}`, () => {
+      const props = {
+        label: 'Labelled button',
+        icon: 'check',
+        theme: 'secondary',
+        size: 'large',
+        extension: 'narrow',
+        className: 'css-class',
+        onClick: jest.fn().mockName('clickHandler'),
+        type: 'reset',
+        [variant]: true
+      }
+
+      it('renders', () => {
+        expect(shallow(<Button />).getElement()).toMatchSnapshot()
+      })
+
+      it('renders with children', () => {
+        expect(
+          shallow(<Button>Test button</Button>).getElement()
+        ).toMatchSnapshot()
+      })
+
+      it('renders with props', () => {
+        expect(shallow(<Button {...props} />).getElement()).toMatchSnapshot()
+      })
+
+      it('renders disabled', () => {
+        expect(
+          shallow(<Button {...props} disabled />).getElement()
+        ).toMatchSnapshot()
+      })
+
+      it('renders busy', () => {
+        expect(
+          shallow(<Button {...props} busy />).getElement()
+        ).toMatchSnapshot()
+      })
+
+      it('renders with Icon', () => {
+        expect(
+          shallow(
+            <Button {...props} icon={<Icon icon="arrow" />} />
+          ).getElement()
+        ).toMatchSnapshot()
```

Comment:
Since the example from the styleguide are automatically snapshotted, why adding more tests ?